src/test/java/jenkins/plugins/okhttp/api/ClientTest.java

```diff
@@ -164,6 +165,40 @@ public void testProxy() throws ExecutionException, InterruptedException, IOExcep
         }
     }
 
+    @Test
+    public void testProxyWithBasicAuthAndRealm() throws ExecutionException, InterruptedException, IOException {
+        jenkinsRule.jenkins.setProxy(new ProxyConfiguration("127.0.0.1", proxy.port(), "proxy-user", "proxy-pass"));
+        secureProxy("Basic realm=\"somerealm\"", "proxy-user:proxy-pass");
+        server.stubFor(WireMock.get("/hello").willReturn(aResponse().proxiedFrom(proxy.baseUrl())));
+
+        final OkHttpClient client = JenkinsOkHttpClient.newClientBuilder(new OkHttpClient()).build();
+        final Request request = new Request.Builder().get().url(server.url("/hello")).build();
+
+        final Response response = new OkHttpFuture<>(client.newCall(request), OkHttpFuture.GET_RESPONSE).get();
+
+        assertEquals(200, response.code());
+        try (final ResponseBody body = response.body()) {
+            assertEquals("Hello from proxy", body.string());
+        }
+    }
+
+    @Test
+    public void testProxyWithPreemptiveAuth() throws ExecutionException, InterruptedException, IOException {
+        jenkinsRule.jenkins.setProxy(new ProxyConfiguration("127.0.0.1", proxy.port(), "proxy-user", "proxy-pass"));
+        secureProxy("OkHttp-Preemptive", "proxy-user:proxy-pass");
```

Comment:
this should not be seen by the proxy server so if this auth scheme is seen does it not imply something is broken?  (in other words this scheme should be internal to the okhttp API only IIUC)

Reply:
Removed